model-optimizer/mo/ops/pooling.py

```diff
@@ -184,3 +198,10 @@ def pool_infer(node: Node):
                                                        ('window', 'input:0'),
                                                        ('spatial_dims', 'input:0'),
                                                        ('dilation', 'input:0')])
+
+    @staticmethod
+    def reverse_infer(node: Node):
+        input_shape = node.in_port(0).data.get_shape()
+        window = node.soft_get('window', None)
+        if input_shape is None and window is not None:
+            node.in_port(0).data.set_shape(undefined_shape_of_rank(len(window)))
```

Comment:
I think we should rather use here `kernel` attribute from the opset.
Also, rank = 2 + len(kernel) because pooling is applied only to spatial dimensions

Reply:
Nope. The same answer as for the Convolution.